zerver/views/invite.py

```diff
@@ -197,7 +209,13 @@ def generate_multiuse_invite_backend(
     ),
     stream_ids: Sequence[int] = REQ(json_validator=check_list(check_int), default=[]),
 ) -> HttpResponse:
+    if not user_profile.can_create_multiuse_invite_to_realm():
+        # Guest users case will not be handled here as it will
+        # be handled by the decorator above.
+        raise JsonableError(_("Insufficient permission"))
+
     check_if_owner_required(invite_as, user_profile)
+    check_if_administrator_required(invite_as, user_profile)
```

Comment:
I think it would be better to combine both these functions for owner and admin and name it accordingly.